Rollbar/RollbarPayloadTruncator.m

```diff
@@ -0,0 +1,271 @@
+    }
+    
+    if (nil == traces) {
+        return TRUE;
+    }
+    
+    [traces enumerateObjectsUsingBlock:^(id item, NSUInteger idx, BOOL *stop) {
+        NSMutableDictionary *exception = [item objectForKey:@"exception"];
+        if (nil != exception) {
+            [exception removeObjectForKey:@"description"];
+            NSMutableString *message = [exception objectForKey:@"message"];
+            if (nil != message && message.length > exeptionMessageLimit) {
+                [exception setObject:[message substringWithRange:NSMakeRange(0, exeptionMessageLimit)] forKey:@"message"];
+            }
+        }
+        NSMutableArray *frames = [item objectForKey:@"frames"];
+        [frames removeObjectsInRange:NSMakeRange(traceFramesLimit, frames.count - traceFramesLimit)];
+    }];
+    
+    return TRUE;
+}
+
++(BOOL)truncatePayload:(NSMutableDictionary*)payload
+          toTotalBytes:(unsigned long) payloadLimit
+ byLimitingStringBytes:(unsigned long)stringBytesLimit {
+    
+    if (![RollbarPayloadTruncator isTruncationNeeded:payload forLimit:payloadLimit]) {
+        return FALSE;  //payload is small enough, no need to truncate further...
+    }
+    
+    [RollbarPayloadTruncator itereateObjectStructure:payload
+                               whileTuncatingStrings:stringBytesLimit
+     ];
+    return TRUE;
+}
+
++(void)itereateObjectStructure:(id)obj whileTuncatingStrings:(unsigned long)stringBytesLimit {
+    
+    if ([obj isKindOfClass:[NSMutableString class]] && [RollbarPayloadTruncator isMutable:obj]) {
+        //truncate the string obj:
+        [obj setString:[RollbarPayloadTruncator truncateString:obj
+                                                  toTotalBytes:stringBytesLimit]
+         ];
+    } else if ([obj isKindOfClass:[NSArray class]]) {
+        //recurse the collection obj's items:
+        [obj enumerateObjectsUsingBlock: ^(id item, NSUInteger idx, BOOL *stop) {
+            [RollbarPayloadTruncator itereateObjectStructure:item
+                                       whileTuncatingStrings:stringBytesLimit
+             ];
+        }];
+    } else if ([obj isKindOfClass:[NSDictionary class]]) {
+        //recurse the collection obj's items:
+        [obj enumerateKeysAndObjectsUsingBlock: ^(id key, id item, BOOL *stop) {
+            //if ([item isKindOfClass:[NSMutableString class]] && ![item respondsToSelector:@selector(setString:)]) {
+            if ([item isKindOfClass:[NSMutableString class]] && ![RollbarPayloadTruncator isMutable:obj]) {
+                NSMutableString *mutableItem = [item mutableCopy];
+                [obj setObject:mutableItem forKey:key];
+                [RollbarPayloadTruncator itereateObjectStructure:mutableItem
+                                           whileTuncatingStrings:stringBytesLimit];
+            }
+            
+            [RollbarPayloadTruncator itereateObjectStructure:item
```

Comment:
Shouldn't this be in an else?

Reply:
correct, that would make it more optimal at runtime. thanks!